scripts/base/protocols/ldap/main.zeek

```diff
@@ -260,8 +263,13 @@ redef record connection += {
 
 #############################################################################
 event zeek_init() &priority=5 {
-  Log::create_stream(LDAP::LDAP_LOG, [$columns=Message, $ev=log_ldap, $path="ldap", $policy=log_policy]);
-  Log::create_stream(LDAP::LDAP_SEARCH_LOG, [$columns=Search, $ev=log_ldap_search, $path="ldap_search", $policy=log_policy_search]);
+  if ( LDAP::enable ) {
```

Comment:
My initial intention was more of a feature flag to opt out of a developing feature and less a convenience mechanism. I do agree that I made this look like other toggles (which might be more of a convenience).

> 

If there's an issue just with one of the transport protocols, the user need to fiddle by hand.

This is all in anticipation of the used parser potentially performing poorly. All analyzers added here share the same parser so I'd expect them to have similar parsing issues. I think a global feature flag would make sense here.

> Analyzers can be re-enabled at runtime, but this won't create the log streams if LDAP::enabled=F was set with the current approach ...
> ...
> Users (and we) should be able to do `redef Analyzer::disabled_a...`

I'd argue that for a feature flag we wouldn't need the ability to toggle it at runtime.

An additional complexity of Spicy analyzers in Zeek today is that `Analyzer::disable_analyzers` is not sufficient to disable them. This disables them on the script side of things, but still e.g., feeds analyzers with input. This is due to how Spicy analyzers are discovered in HLTO files and loaded via a [separate registration mechanism](https://github.com/zeek/zeek/blob/4af7d7027feb6af88b463443482bf2fdf18ac488/src/spicy/manager.cc#L437-L460) (this is also what e.g., [`Spicy::disable_protocol_analyzer` and similar functions](https://github.com/zeek/zeek/blob/4af7d7027feb6af88b463443482bf2fdf18ac488/src/spicy/manager.cc#L437-L460) interact with).

What we will ultimately need to fully support `Analyzer::disabled_analyzers` for Spicy-managed analyzers would be the functionality in `spicy::Manager` which can inspect that value at runtime and react on it. I didn't really plan to address that with this patch, but I can look into it.

Reply:
> My initial intention was more a feature flag to opt out of a developing feature and less a convenience mechanism. I do agree that I made this look like other toggles (which might be more of a convenience).

Yep, got it. I still think the following is all that should be needed to disable the new LDAP analyzers. We can document this as receipt in NEWS in case there are issues.
```
redef Analyzer::disabled_analyzers += {
  Analyzer::ANALYZER_LDAP_TCP,
  Analyzer::ANALYZER_LDAP_UDP,
```

The other side might be an expectation of `::enable` flags for other modules/analyzers.


> An additional complexity of Spicy analyzers in Zeek today is...
> What we will ultimately need to fully support ...

I actually thought that is functional and skimming the code, `toggleProtocolAnalyzer()` fowards to `analyzer_mgr->Enable/Disable()` which is all that `Analyzer::disable_analyzer()` or the `disabled_analyzers` mechanism should be doing (modulo the replaces cases). Have you tried? -I have not ;-)

Reply:
I tried and you are right, this works. Either I had a completely wrong model, or something changed here. I am removing the flag and will leave it up to users to just disable these analyzers.